tests/functional/requirements.txt

```diff
@@ -1,4 +1 @@
-# juju: Needs to be below 3.1 to maintain compatibility with Juju controller < 3.0
-# juju: Latest release in 2.9.x seems to contain bug that hangs forever when awaiting
-# Action.wait()
-juju >3.0,<3.1
+juju >= 3.1
```

Comment:
I don't think specifying package version for juju is needed. Leaving it as default would be better IMO because we don't increase future maintenance burden.